games/ludii/jni_utils.cc

```diff
@@ -54,19 +61,40 @@ void JNIUtils::InitJVM(std::string jar_location) {
     return;
   }
 
+  //#define CHECK_JNI  // Uncomment this to run extra checks for JNI
+
 #ifdef JNI_VERSION_1_2
   JavaVMInitArgs vm_args;
-  JavaVMOption options[1];
+
+#ifdef CHECK_JNI
+  const size_t num_jvm_args = 2;
+#else
+  const size_t num_jvm_args = 1;
+#endif
+
+  JavaVMOption options[num_jvm_args];
   std::string java_classpath = "-Djava.class.path=" + jar_location;
   char* c_classpath = strdup(java_classpath.c_str());
   options[0].optionString = c_classpath;
+
+#ifdef CHECK_JNI
+  std::string check_jni = "-Xcheck:jni";
+  char* c_check_jni = strdup(check_jni.c_str());
```

Comment:
This strdup/free is pretty ugly - is there a reason you can't just use check_jni.c_str() directly?

Reply:
It's because `c_str()` returns a `const char*`, but in this JavaVMOption type of JNI they require a non-const `char*`.

Reply:
That's very strange. You can try using .data() instead of c_str to get a non-const pointer.

Reply:
That's very strange. You can try using .data() instead of c_str to get a non-const pointer.

Reply:
That one's also const isn't it? I could just cast it to a non-const pointer... but I'm not sure that that's less ugly than the current implementation?

Reply:
Since C++17 (which we target), there is a non-const .data(), so please try it.
You can also `const_cast<char*>(check_jni.c_str())`, which imo is prettier than the strdup stuff.
Also please do the same for c_classpath above